sdk/core/core-rest-pipeline/src/nodeHttpClient.ts

```diff
@@ -244,28 +245,31 @@ class NodeHttpClient implements HttpClient {
   }
 
   private getOrCreateAgent(request: PipelineRequest, isInsecure: boolean): http.Agent {
-    if (!request.disableKeepAlive) {
-      if (isInsecure) {
-        if (!this.httpKeepAliveAgent) {
-          this.httpKeepAliveAgent = new http.Agent({
-            keepAlive: true,
-          });
-        }
+    if (!request.tlsSettings && request.disableKeepAlive) {
+      return isInsecure ? http.globalAgent : https.globalAgent;
+    }
+    const settingsIndex = request.tlsSettings ?? {};
+    let agent: http.Agent | undefined = isInsecure
+      ? this.httpKeepAliveAgent
+      : this.httpsKeepAliveAgent?.get(settingsIndex);
```

Comment:
curious why this is `?.get` and not `.get`

Reply:
You are right, no need to ?.get